sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/pubsub/PubsubJsonTableProvider.java

```diff
@@ -54,125 +52,27 @@ public String getTableType() {
-     * PubsubIO.Read.Builder#withTimestampAttribute}.
-     *
-     * <p>Short version: it has to be either millis since epoch or string in RFC 3339 format.
-     *
-     * <p>If the attribute is specified then event timestamps will be extracted from the specified
-     * attribute. If it is not specified then message publish timestamp will be used.
-     */
-    @Nullable
-    abstract String getTimestampAttribute();
-
-    /**
-     * Optional topic path which will be used as a dead letter queue.
-     *
-     * <p>Messages that cannot be processed will be sent to this topic. If it is not specified then
-     * exception will be thrown for errors during processing causing the pipeline to crash.
-     */
-    @Nullable
-    abstract String getDeadLetterQueue();
-
-    /**
-     * Pubsub topic name.
-     *
-     * <p>Topic is the only way to specify the Pubsub source. Explicitly specifying the subscription
-     * is not supported at the moment. Subscriptions are automatically created (but not deleted).
-     */
-    abstract String getTopic();
-
-    /**
-     * Table schema, describes Pubsub message schema.
-     *
-     * <p>If {@link #getUseFlatSchema()} is not set, schema must contain exactly fields
-     * 'event_timestamp', 'attributes, and 'payload'. Else, it must contain just 'event_timestamp'.
-     * See {@linkA PubsubMessageToRow} for details.
-     */
-    public abstract Schema getSchema();
-
-    static Builder builder() {
-      return new AutoValue_PubsubJsonTableProvider_PubsubIOTableConfiguration.Builder();
-    }
-
-    @AutoValue.Builder
-    abstract static class Builder {
-      abstract Builder setUseFlatSchema(boolean useFlatSchema);
-
-      abstract Builder setSchema(Schema schema);
-
-      abstract Builder setTimestampAttribute(String timestampAttribute);
-
-      abstract Builder setDeadLetterQueue(String deadLetterQueue);
-
-      abstract Builder setTopic(String topic);
-
-      abstract PubsubIOTableConfiguration build();
+  public BeamSqlTable buildBeamSqlTable(Table tableDefinition) {
+    JSONObject tableProperties = tableDefinition.getProperties();
+    PubsubSchemaCapableIOProvider ioProvider = new PubsubSchemaCapableIOProvider();
+
+    try {
+      RowJsonDeserializer deserializer =
+          RowJsonDeserializer.forSchema(ioProvider.configurationSchema())
+              .withMissingFieldBehavior(RowJsonDeserializer.MissingFieldBehavior.ALLOW_MISSING);
+
+      Row configurationRow =
+          newObjectMapperWith(deserializer).readValue(tableProperties.toString(), Row.class);
+
+      SchemaIO pubsubSchemaIO =
+          ioProvider.from(
+              tableDefinition.getLocation(), configurationRow, tableDefinition.getSchema());
+
+      return PubsubIOJsonTable.fromSchemaIO(pubsubSchemaIO);
+    } catch (InvalidConfigurationException | InvalidSchemaException e) {
+      throw new InvalidTableException(e.getMessage());
+    } catch (JsonProcessingException e) {
+      throw new AssertionError();
```

Comment:
This should have a meaningful message like "Failed to re-parse TBLPROPERTIES JSON"